kaggle_environments/envs/halite/helpers.py

```diff
@@ -16,138 +16,12 @@
-        """
-        Converts a 2d position in the form (x, y) to an index in the observation.halite list.
-        See index_to_position for the inverse.
-        """
-        return (size - self.y - 1) * size + self.x
-
-    @staticmethod
-    def from_index(index: int, size: int) -> 'Point':
-        """
-        Converts an index in the observation.halite list to a 2d position in the form (x, y).
-        See position_to_index for the inverse.
-        """
-        y, x = divmod(index, size)
-        return Point(x, (size - y - 1))
-
-    def __abs__(self) -> 'Point':
-        return self.map(operator.abs)
-
-    def __add__(self, other: Union[Tuple[int, int], 'Point']) -> 'Point':
-        return self.map2(other, operator.add)
-
-    def __eq__(self, other: Union[Tuple[int, int], 'Point']) -> bool:
-        try:
-            return self[0] == other[0] and self[1] == other[1]
-        except (TypeError, IndexError):
-            return False
-
-    def __floordiv__(self, denominator: int) -> 'Point':
-        return self.map(lambda x: x // denominator)
-
-    def __hash__(self) -> int:
-        return hash((self.x, self.y))
-
-    def __mod__(self, mod: int) -> 'Point':
-        return self.map(lambda x: x % mod)
-
-    def __mul__(self, factor: int) -> 'Point':
-        return self.map(lambda x: x * factor)
-
-    def __neg__(self) -> 'Point':
-        return self.map(operator.neg)
-
-    def __str__(self):
-        return f"({self.x}, {self.y})"
-
-    def __sub__(self, other: Union[Tuple[int, int], 'Point']) -> 'Point':
-        return self.map2(other, operator.sub)
-
-
-TElement = TypeVar('TElement')
-THash = TypeVar('TComparable')
-
-
-def group_by(elements: Iterable[TElement], selector: Callable[[TElement], THash]) -> Dict[THash, List[TElement]]:
-    results = {}
-    for element in elements:
-        key = selector(element)
-        if key not in results:
-            results[key] = []
-        results[key].append(element)
-    return results
-
-
-TKey = TypeVar('TKey')
-TValue = TypeVar('TValue')
-
-
-class ReadOnlyDict(Generic[TKey, TValue]):
-    """Offers Dict-like semantics while preventing modification of the underlying datastructure."""
-    def __init__(self, data: Union['ReadOnlyDict[TKey, TValue]', Dict[TKey, TValue]]):
-        if isinstance(data, dict):
-            self._data = data
-        else:
-            # If it's not a Dict it must be a ReadOnlyDict based on our type
-            # Unwrap inner ReadOnlyDict's data
-            self._data = data._data
-
-    def __getitem__(self, item) -> Optional[TValue]:
-        return self._data.get(item)
-
-    def __iter__(self):
-        return self._data.__iter__()
-
-    def __str__(self):
-        return self._data.__str__()
-
-    def keys(self):
-        return self._data.keys()
-
-    def values(self):
-        return self._data.values()
-# endregion
-
-
 # region Data Model Classes
-class Observation(ReadOnlyDict[str, any]):
+class Observation(Observation):
```

Comment:
This is a bit weird to see (like is it subclassing itself via some Python circular black magic??) and stems from the wildcard import, might be nicer to read `class Observation(base_helpers.Observation)` or something via `import .. as ..`